desktop/src/app.rs

```diff
@@ -449,6 +453,46 @@ impl App {
                 _ => (),
             }
 
+            if let Some(Event { event, .. }) = gilrs.as_mut().and_then(|gilrs| gilrs.next_event()) {
+                fn button_to_gamepad_button(button: Button) -> Option<GamepadButton> {
+                    match button {
+                        Button::South => Some(GamepadButton::South),
+                        Button::East => Some(GamepadButton::East),
+                        Button::North => Some(GamepadButton::North),
+                        Button::West => Some(GamepadButton::West),
+                        Button::LeftTrigger => Some(GamepadButton::LeftTrigger),
+                        Button::LeftTrigger2 => Some(GamepadButton::LeftTrigger2),
+                        Button::RightTrigger => Some(GamepadButton::RightTrigger),
+                        Button::RightTrigger2 => Some(GamepadButton::RightTrigger2),
+                        Button::Select => Some(GamepadButton::Select),
+                        Button::Start => Some(GamepadButton::Start),
+                        Button::DPadUp => Some(GamepadButton::DPadUp),
+                        Button::DPadDown => Some(GamepadButton::DPadDown),
+                        Button::DPadLeft => Some(GamepadButton::DPadLeft),
+                        Button::DPadRight => Some(GamepadButton::DPadRight),
+                        _ => None,
+                    }
+                }
+
+                match event {
```

Comment:
how about writing it in the same way as above, with 2 matches?
The nested function wouldn't be needed then.

(I noticed that these two blocks of code are super similar, but are written in different styles - no strong preference between them though)

Reply:
I didn't really like the approach with two nested matches, one which requires unreachable!, so I would rather not do that.